### ISSUE TITLE
usefulness of clustering for smaller numbers of ROIs (~50-100)

### ISSUE THREAD

**Issue by torreydactyl** [zebrain-lab/Toolbox-Romano-et-al#3]
Hi Sebastian et al,

I came across your pre-print and associated code because I'm trying to group neurons into clusters from in vivo Xenopus tadpole calcium imaging. However, my data has 50-100 neurons per tadpole dataset that respond to my stimuli. Would the detection of neuronal assemblies module work on this number of neurons? I already have df/f traces for all ROIs and would just need to format them for input into the module.  

Thanks,
Torrey Truszkowski

**Comment by sebromano** [comment]
Hi Torrey,

in principle, clustering algorithms suffer when data dimensionality is too high, not too low. However, you still need to sample from enough neurons for a cluster to have any biological sense. So, it will depend on the dataset. If you have ROIs that are significantly correlated, and you have a fairly long recording, you should be able to detect clusters. Anyway, the program will evaluate the significance of the clusters by controlling against surrogate shuffled datasets. 
For sparse calcium activity and 500 ROIs, I obtained stable clusters when recording for at least 10 minutes imaged at 4Hz. But that could change a lot depending on the level of activity in the data.
As a rule of thumb, to evaluate your confidence on the detected clusters, you could split your dataset in two (e.g., for a 20 min dataset split it into two 10 min blocks) run the algorithm on one of the blocks and see how the detected clusters explain the correlation structure of the remaining block. A simple and visual test would be to calculate the correlation matrix of the ROIs in the second block, and sort this matrix according to clusters found in the first block: you should observe a matrix with a diagonal-like organization.
In conclusion, I would definitively encourage you to apply the clustering procedure on the data, there is no reason not to try. But clustering results should always be evaluated with care.
Best,
Sebastian.

**Comment by torreydactyl** [comment]
Hi Sebastian,

Thanks for the information. I'm working through the instructions in the preprint to run my data through the Detection of Neural Assemblies methods. However, I'm getting an error when running it that seems to indicate a problem with dataAllCells.avg. Mine is a 250x250 unit16, which I thought would work, but I get this error while running the heirarchachal clustering algorithm:
Undefined function or variable "dataAllCells".

Error in FindAssemblies (line 403)
    numPlanes=size(dataAllCells.avg,3);

Any ideas? Thanks!

![image](https://user-images.githubusercontent.com/21325236/29723923-784b1a66-8994-11e7-9d0b-ec757c786619.png)

**Comment by sebromano** [comment]
Hi Torrey,

first of all, sorry for the delayed response.
Indeed, there was a bug when importing fluorescence data to the toolbox, which I fixed. Thank you for reporting it. 
You can download the new version of FindAssemblies.m and it should work now.
Let me know if you have any further problems.
Best, 
Sebastian.
